.circleci/config.yml

```diff
@@ -1,147 +1,21 @@
-          command: npm i --save-dev jest-junit
-      - run:
-          name: Run tests with JUnit as reporter
-          # command: jest --ci --runInBand --reporters=default --reporters=jest-junit
-          command: npm run test:ci
-          environment:
-            JEST_JUNIT_OUTPUT_DIR: ./reports/
-      - store_test_results:
-          path: ./reports/
-      - store_artifacts:
-          path: ./reports/
-
-  smoke_test:
-    machine:
-      image: ubuntu-2204:2024.01.1
-    steps:
-      - checkout
-      - attach_workspace:
-          at: ./
-      - bats/install
-      - run:
-          name: What's the BATS?
-          command: |
-            which bats
-            bats --version
-      - cypress/install
-      - run:
-          name: What's the Cypress?
-          command: npx cypress --version
-      - run:
-          name: Spin up and run e2e smoke tests
-          command: make smoke
-      - store_test_results:
-          path: ./smoke-tests/
-      - store_artifacts:
-          path: ./smoke-tests/report.xml
-      - store_artifacts:
-          path: ./smoke-tests/collector/data-results
-      - run:
-          name: Tear down e2e smoke tests
-          command: make unsmoke
-
-  publish_github:
-    executor: github
-    steps:
-      - run:
-          name: "GHR Draft"
-          command: ghr -draft -n ${CIRCLE_TAG} -t ${GITHUB_TOKEN} -u ${CIRCLE_PROJECT_USERNAME} -r ${CIRCLE_PROJECT_REPONAME} -c ${CIRCLE_SHA1} ${CIRCLE_TAG}
+  path-filtering: circleci/path-filtering@1.1.0
 
-  publish_npm:
-    executor: node
-    steps:
-      - checkout
-      - run:
-          name: store npm auth token
-          command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/.npmrc
-      - run: npm ci
-      - run: npm run build
-      - run: npm publish --access public
+parameters:
+  base-path:
+    type: string
+    default: ''
 
 workflows:
-  build:
+  generate-config:
     jobs:
-      - check-format:
-          <<: *filters_always
-      - lint:
-          <<: *filters_always
-      - test:
-          <<: *filters_always
-      - build:
-          <<: *filters_always
-          requires:
-            - test
-      - smoke_test:
-          <<: *filters_always
-          requires:
-            - build
-      - publish_github:
-          <<: *filters_publish
-          context: Honeycomb Secrets for Public Repos
-          requires:
-            - build
-            - smoke_test
-      - publish_npm:
-          <<: *filters_publish
-          context: Honeycomb Secrets for Public Repos
-          requires:
-            - build
-            - smoke_test
+      - path-filtering/filter:
+          base-revision: main
+          config-path: .circleci/build-npm-package.yml
+          mapping: |
+            packages/honeycomb-opentelemetry-web/.* package-name "honeycomb-opentelemetry-web"
+            packages/honeycomb-react-user-instrumentation/.* package-name "honeycomb-react-user-instrumentation"
```

Comment:
from the group review: how difficult will it be for us to trigger builds for _all_ things that have changed

in the ideal world:
- dev pushes some changes
- circleci / npm / tooling stuff figures out what changed, and then:
  - runs tests for changed stuff
  - builds artifacts for changed stuff
  - creates a release for changed stuff